### PR TITLE
User Preferences: Fix `auth_callback` permission check

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Fix user meta permission check if editing another user. 

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: includes/User/Preferences.php [hunk 1]
```diff
@@ -112,9 +112,16 @@ public function register() {
 	/**
 	 * Auth callback.
 	 *
+	 * @since 1.4.0
+	 *
+	 * @param bool   $allowed Unused.
+	 * @param string $meta_key Unused.
+	 * @param int    $user_id User id of user meta.
+	 * @param int    $current_user_id Current user id.
```
Comment (swissspidy) [inline includes/User/Preferences.php]: just some minor docs suggestions.

```suggestion
	 * @param bool   $allowed Unused. Whether the user can add the object meta.
	 * @param string $meta_key Unused. The meta key.
	 * @param int    $user_id ID of the user being edited.
	 * @param int    $current_user_id The currently editing user's ID.
```

I usually tend to keep the original variable names (e.g. `$object_id` & `$user_id`) but I suppose this makes it more understandable

----

Note: need to fix docblock alignment either way.

File: tests/phpunit/tests/User/Preferences.php [hunk 2]
```diff
@@ -151,6 +151,41 @@ public function test_enables_author_user_to_update_meta_field() {
 		$this->assertEqualSets( [ 'hello' => 'world' ], $data['meta'][ \Google\Web_Stories\User\Preferences::ONBOARDING_META_KEY ] );
 	}
 
+	/**
+	 * @covers ::register
+	 * @covers ::can_edit_current_user
+	 */
+	public function test_permission_check_for_authors() {
```
Comment (swissspidy) [inline tests/phpunit/tests/User/Preferences.php:158]: 👍

**Review by swissspidy — APPROVED** [review]
LGTM, just one nit

File: includes/User/Preferences.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/User/Preferences.php]: > I usually tend to keep the original variable names (e.g. $object_id & $user_id) but I suppose this makes it more understandable

Me too, it felt wrong here....

**Comment by spacedmonkey** [comment]
Fixed comment.

File: includes/User/Preferences.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/User/Preferences.php]: Fixed.

**Pull request merged**